test/promql/tests/test_unschedulable_pods.yaml

```diff
@@ -0,0 +1,44 @@
+# This is the main input for unit testing.
+# Only this file is passed as command line argument
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  # Interval for the samples of the series
+  - interval: 1m
+    # Series data
+    input_series:
+
+      # Initially pod is unscheduled but later status changed to scheduled
+      - series: 'kube_pod_status_unschedulable{namespace="prod", pod="prod-pod-1"}'
+        values: '1x40 0x20'
+
+      - series: 'kube_pod_status_unschedulable{namespace="prod-test", pod="prod-pod-2"}'
+        values: '1x60'
+
```

Comment:
Can we also add a test case in which the pod/namespace qualify for the alert, but it has a window of reachability within a long-enough unreachable status?

Something like this:

```
      - series: 'kube_pod_status_unschedulable{namespace="prod-test", pod="prod-pod-2"}'
        values: '1x29 0 1x28'
```